Packages/com.unity.inputsystem/InputSystem/Editor/InputActionAsset/InputActionAssetManager.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.IO;
+using UnityEditor;
+
+namespace UnityEngine.Experimental.Input.Editor
+{
+    [Serializable]
+    public class InputActionAssetManager
+    {
+        [SerializeField]
+        internal InputActionAsset m_AssetObjectForEditing;
+        [SerializeField]
+        private InputActionAsset m_ImportedAssetObject;
+        [SerializeField]
+        private string m_AssetGUID;
+        [SerializeField]
+        private string m_AssetPath;
+        [SerializeField]
+        private string m_ImportedAssetJson;
+        [SerializeField]
+        private bool m_IsDirty;
+
+        private SerializedObject m_SerializedObject;
+        Action<bool> m_SetTitle;
+
+        InputActionAsset importedAsset
+        {
+            get
+            {
+                if (m_ImportedAssetObject == null)
+                {
+                    LoadImportedObjectFromGuid();
+                }
+                return m_ImportedAssetObject;
+            }
+        }
+
+        public InputActionAssetManager(InputActionAsset inputActionAsset)
+        {
+            m_ImportedAssetObject = inputActionAsset;
+            m_AssetPath = AssetDatabase.GetAssetPath(importedAsset);
+            m_AssetGUID = AssetDatabase.AssetPathToGUID(m_AssetPath);
+        }
+
+        public SerializedObject serializedObject
+        {
+            get { return m_SerializedObject; }
+        }
+
+        public bool dirty
+        {
+            get { return m_IsDirty; }
+        }
+
+        public void InitializeObjectReferences()
+        {
+            if (m_AssetObjectForEditing == null)
+            {
+                CreateWorkingCopyAsset();
+            }
+            m_SerializedObject = new SerializedObject(m_AssetObjectForEditing);
+        }
+
+        internal void CreateWorkingCopyAsset()
+        {
+            if (m_AssetObjectForEditing != null)
+            {
+                UnloadAssets();
+            }
+            // Duplicate the asset along 1:1. Unlike calling Clone(), this will also preserve
+            // GUIDs.
+            m_AssetObjectForEditing = Object.Instantiate(importedAsset);
+            m_AssetObjectForEditing.hideFlags = HideFlags.HideAndDontSave;
+            m_AssetObjectForEditing.name = importedAsset.name;
+            m_SerializedObject = new SerializedObject(m_AssetObjectForEditing);
+        }
+
+        public void UnloadAssets()
```

Comment:
Minor but this isn't really unloading something. It's just destroying a transient object.